src/ArduinoPackageManager.ts

```diff
@@ -50,15 +41,24 @@ export class ArduinoPackageManager {
     }
   }
 
-  static async installBoard(boardId: string) {
-    const board = BOARD_INFO[boardId];
-    if (!board) {
+  static async installBoard(context: vscode.ExtensionContext, boardId: string) {
+    const boardList = context.asAbsolutePath(
+        path.join(FileNames.resourceFolderName, FileNames.boardListFileName));
+    const boardsJson = require(boardList);
+
+    const board = boardsJson.boards.filter((template: Board) => {
+      return template.id === boardId;
+    });
+
+    if (!board || !board[0]) {
       return;
     }
 

```

Comment:
check if installation is undefined here, and return if the board doesn't have installation property.